awswrangler/s3/_write_parquet.py

```diff
@@ -593,8 +593,11 @@ def to_parquet(  # pylint: disable=too-many-arguments,too-many-locals,too-many-b
             commit_trans = True
 
     df = _apply_dtype(df=df, dtype=dtype, catalog_table_input=catalog_table_input, mode=mode)
+
+    # Index of DataFrame cannot be used when using Ray and Modin
+    parse_schema_with_index = index and engine.get() != EngineEnum.RAY
     schema: pa.Schema = _data_types.pyarrow_schema_from_pandas(
-        df=df, index=index, ignore_cols=partition_cols, dtype=dtype
```

Comment:
I'm not happy with this to be honest. Any suggestions are welcome. I wonder if we should actually just throw an error when a customer tries to write an indexed parquet file from distributed Wrangler. If I write a file and say I want an index, I'd rather the library tell me it can't do it, rather than silently failing. Especially since this implies a change in behaviour from non-distributed Wrangler.

Reply:
I am not sure I understand the root of the problem here. What is the error that you are seeing in the tests that lead to introduce this? AFAIK, @kukushking did not see an issue with parquet tests when writing the index, even in distributed mode, so curious to understand what is specific to Athena here.